jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/DBRecordWritable.java

```diff
@@ -61,16 +60,30 @@ public void write(PreparedStatement statement) throws SQLException {
     if (columnValues == null) {
       throw new SQLException("No data available to be written");
     }
-    ParameterMetaData parameterMetaData = statement.getParameterMetaData();
     for (int i = 0; i < columnValues.length; i++) {
       Object value = columnValues[i];
-      if ((parameterMetaData.getParameterType(i + 1) == Types.CHAR) && value != null && value instanceof Boolean) {
+      if ((checkParamMeta(statement) && statement.getParameterMetaData().getParameterType(i + 1) == Types.CHAR)
+          && value != null && value instanceof Boolean) {
         value = ((Boolean) value).booleanValue() ? "1" : "0";
       }
       statement.setObject(i + 1, value);
     }
   }
 
+  private Boolean checkParamMeta(PreparedStatement statement) throws SQLException {
+    String dbType = statement.getConnection().toString().split(":")[1].toUpperCase();
```

Comment:
Relying on `getConnection().toString()` is risky since the result can change at any time without notice. Moreover, I am not sure if the snippet here is gonna work for every supported database.

Reply:
Yes, i agree this change is not very elegant. I have moved the code logic into `jdbcSerDe:initialize`. Please check again.